solidity/ecdsa/contracts/libraries/DKG.sol

```diff
@@ -423,6 +423,20 @@ library DKG {
         return (maliciousResultHash, maliciousSubmitter);
     }
 
+    /// @notice Checks if DKG result is valid for the current DKG.
+    /// @param result DKG result.
+    /// @return True if the result is valid. If the result is invalid it returns
+    ///         false and an error message.
+    function isResultValid(Data storage self, Result calldata result)
+        external
+        view
+        returns (bool, string memory)
+    {
+        require(self.startBlock > 0, "DKG has not been started");
```

Comment:
`self.startBlock > 0` is concise, but I think `require(currentState(self) == State.AWAITING_RESULT || currentState(self) == State.CHALLENGE)` is a bit more descriptive. I'm okay with both though.